types/victory/index.d.ts

```diff
@@ -2936,11 +2936,12 @@ declare module 'victory' {
          */
         radius?: number;
         /**
-         * When creating a donut chart, this prop determines the number of pixels between
-         * the center of the chart and the inner edge of a donut. When this prop is set to zero
-         * a regular pie chart is rendered.
+         * The `innerRadius` prop determines the number of pixels between the center of the chart
+         * and the inner edge of a donut chart. When this prop is set to zero a regular pie chart is rendered.
+         * When this prop is given as a function, `innerRadius` will be evaluated for each slice
+         * of the pie with the props corresponding to that slice
          */
-        innerRadius?: number;
+        innerRadius?: number | ((props: VictoryPieProps) => number);
```

Comment:
I'm not actually sure about this change :thinking:, as the documentation states:
> When this prop is given as a function, `innerRadius` will be evaluated for each slice of the pie with the props corresponding to that slice

Maybe we should match the types from the `Slice` [component](https://github.com/FormidableLabs/victory/blob/9f0f3cae1ebaeb889eeb20b289413f40e03ac19f/packages/victory-pie/src/slice.js#L54)?

Reply:
WDYT @peterblazejewicz?

Reply:
@ibarrae 
thanks for checking this! ok, could be that way (taking default props, addign one from Slice and passign as props)?
```diff
--- a/types/victory/index.d.ts
+++ b/types/victory/index.d.ts
@@ -108,6 +108,8 @@ declare module 'victory' {
     type StringOrNumberOrCallback = string | number | VictoryStringOrNumberCallback;
     type NumberOrCallback = number | VictoryNumberCallback;
 
+    type SliceNumberOrCallback<T, P = null> = number | ((props: Omit<T, P>) => number);
+
     type VictoryStyleObject = { [K in keyof React.CSSProperties]: StringOrNumberOrCallback };
     /**
      * Style interface used in components/themeing
@@ -2936,11 +2938,12 @@ declare module 'victory' {
          */
         radius?: number;
         /**
-         * When creating a donut chart, this prop determines the number of pixels between
-         * the center of the chart and the inner edge of a donut. When this prop is set to zero
-         * a regular pie chart is rendered.
+         * The `innerRadius` prop determines the number of pixels between the center of the chart
+         * and the inner edge of a donut chart. When this prop is set to zero a regular pie chart is rendered.
+         * When this prop is given as a function, `innerRadius` will be evaluated for each slice
+         * of the pie with the props corresponding to that slice
          */
-        innerRadius?: number;
+        innerRadius?: number | ((props: VictorySliceProps) => number);
         /**
          * Set the cornerRadius for every dataComponent (Slice by default) within VictoryPie
          */
@@ -3018,4 +3021,70 @@ declare module 'victory' {
          */
         groupComponent?: React.ReactElement;
     }
+
+    export interface VictorySliceProps extends VictoryCommonProps {
+        /**
+         * the corner radius to apply to this slice.
+         * When this prop is given as a function
+         * it will be called with the rest of the props supplied to Slice.
+         */
+        cornerRadius?: SliceNumberOrCallback<VictorySliceProps, 'cornerRadius'>;
+        /**
+         * the data point corresponding to this slice
+         */
+        datum?: object;
+        /**
+         * the inner radius of the slice.
+         * When this prop is given as a function
+         * it will be called with datum and active.
+         */
+        innerRadius?: number | ((props: {
+            active?: boolean,
+            datum?: object,
+        }) => number);
+        /**
+         * the angular padding to add to the slice.
+         * When this prop is given as a function it will be called with
+         * the rest of the props supplied to Slice.
+         */
+        padAngle?: SliceNumberOrCallback<VictorySliceProps, 'padAngle'>;
+        /**
+         * the rendered path element
+         * @default pathComponent={<Path/>}
+         */
+        pathComponent?: React.ReactElement;
+        /**
+         * a function that calculates the path of a given slice.
+         * When given, this prop will be called with the slice object
+         */
+        pathFunction?: (props: VictorySliceProps) => string;
+        /**
+         * the outer radius of the slice.
+         * When this prop is given as a function it will be called with
+         * the rest of the props supplied to Slice.
+         */
+        radius?: SliceNumberOrCallback<VictorySliceProps, 'radius'>;
+        /**
+         * an object specifying the `startAngle`, `endAngle`, `padAngle`,
+         * and `data` of the slice
+         */
+        slice: {
+            startAngle?: number;
+            endAngle?: number;
+            padAngle?: number;
+            data?: any[];
+        };
+        /**
+         * the end angle the slice.
+         * When this prop is given as a function it will be called
+         * with the rest of the props supplied to Slice.
+         */
+        sliceEndAngle?: SliceNumberOrCallback<VictorySliceProps, 'sliceEndAngle'>;
+        /**
+         * the start angle the slice.
+         * When this prop is given as a function it will be called
+         * with the rest of the props supplied to Slice
+         */
+        sliceStartAngle?: SliceNumberOrCallback<VictorySliceProps, 'sliceStartAngle'>;
+    }
 }
--- a/types/victory/victory-tests.tsx
+++ b/types/victory/victory-tests.tsx
@@ -25,6 +25,7 @@ import {
     VictoryClipContainer,
     VictoryPortal,
     VictoryStringOrNumberCallback,
+    VictorySliceProps,
 } from 'victory';
 
 const commonData1 = [
@@ -661,6 +662,7 @@ test = (
             { animal: 'Bird', pet: 15, wild: 40 },
         ]}
         x={'animal'}
+        innerRadius={(props: VictorySliceProps) => 2 }
         y={data => data.pet + data.wild}
     />
 );
@@ -960,3 +962,21 @@ const fullTheme: RecursiveRequired<Required<VictoryThemeDefinition>> = {
         width: 0,
     },
 };
+
+// Slice.props
+const sliceProps: VictorySliceProps = {
+    cornerRadius: props => props.cornerRadius, // $ExpectError
+    datum: { x: 'Cat', y: 62 },
+    innerRadius: props => props.innerRadius, // $ExpectError
+    padAngle: props => props.padAngle, // $ExpectError
+    pathFunction: sliceProps => 'M1,1',
+    radius: props => props.radius, // $ExpectError
+    slice: {
+        data: [],
+        endAngle: 0,
+        padAngle: 0,
+        startAngle: 0,
+    },
+    sliceEndAngle: props => props.sliceEndAngle, // $ExpectError
+    sliceStartAngle: props => props.slieStartAngle, // $ExpectError
+};
;
```

Reply:
That looks really nice! Great work :tada:

Reply:
Let me know when you push changes on this so I can make a final review :smile:

Reply:
done, rebased and squashed. Thanks for taking time to look into this, appreciated!